File: src/main/java/physic/ai/domain/products/dto/ProductMapper.java
package physic.ai.domain.products.dto;

import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;
import physic.ai.domain.products.ProductEntity;

@Mapper(componentModel = "cdi",
        unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface ProductMapper {
    ProductEntity toProductEntity(NewProductDto newProduct);
}

File: src/main/java/physic/ai/application/dto/ProductMapper.kt
package physic.ai.application.dto

import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy
import physic.ai.domain.model.Product
import physic.ai.infrastructure.persistency.entities.ProductEntity


@Mapper(componentModel = "cdi", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface ProductMapper {
    fun toProductEntity(newProduct: NewProductDto): ProductEntity
    fun toProduct(newProduct: NewProductDto): Product
}
